internal/provider/resource_aws_transit_gateway_attachment.go

```diff
@@ -0,0 +1,319 @@
+
+		return diag.Errorf("unable to check for presence of an existing HVN (%s): %v", hvnID, err)
+	}
+	log.Printf("[INFO] HVN (%s) found, proceeding with create", hvnID)
+
+	// Check if TGW attachment already exists
+	_, err = clients.GetTGWAttachmentByID(ctx, client, tgwAttachmentID, hvnID, loc)
+	if err != nil {
+		if !clients.IsResponseCodeNotFound(err) {
+			return diag.Errorf("unable to check for presence of an existing Transit gateway attachment (%s): %v", tgwAttachmentID, err)
+		}
+
+		log.Printf("[INFO] Transit gateway attachment (%s) not found, proceeding with create", tgwAttachmentID)
+	} else {
+		return diag.Errorf("a Transit gateway attachment with transit_gateway_attachment_id=%s, hvn_id=%s and project_id=%s already exists - to be managed via Terraform this resource needs to be imported into the state. Please see the resource documentation for hcp_transit_gateway_attachment for more information", tgwAttachmentID, hvnID, loc.ProjectID)
+	}
+
+	// Create TGW attachment
+	createTGWAttachmentParams := network_service.NewCreateTGWAttachmentParams()
+	createTGWAttachmentParams.HvnID = hvnID
+	createTGWAttachmentParams.HvnLocationOrganizationID = loc.OrganizationID
+	createTGWAttachmentParams.HvnLocationProjectID = loc.ProjectID
+	createTGWAttachmentParams.Body = &networkmodels.HashicorpCloudNetwork20200907CreateTGWAttachmentRequest{
+		Cidrs: destinationCIDRs,
+		Hvn: &sharedmodels.HashicorpCloudLocationLink{
+			ID:       hvnID,
+			Location: loc,
+		},
+		ID: tgwAttachmentID,
+		ProviderData: &networkmodels.HashicorpCloudNetwork20200907CreateTGWAttachmentRequestProviderData{
+			AwsData: &networkmodels.HashicorpCloudNetwork20200907AWSCreateRequestTGWData{
+				ResourceShareArn: resourceShareARN,
+				TgwID:            tgwID,
+			},
+		},
+	}
+	log.Printf("[INFO] Creating Transit gateway attachment for HVN (%s) and Transit gateway (%s)", hvnID, tgwID)
+	createTGWAttachmentResponse, err := client.Network.CreateTGWAttachment(createTGWAttachmentParams, nil)
+	if err != nil {
+		return diag.Errorf("unable to create Transit gateway attachment for HVN (%s) and Transit gateway (%s): %v", hvnID, tgwID, err)
+	}
+
+	tgwAtt := createTGWAttachmentResponse.Payload.TgwAttachment
+
+	// Wait for TGW attachment creation to complete
+	if err := clients.WaitForOperation(ctx, client, "create Transit gateway attachment", loc, createTGWAttachmentResponse.Payload.Operation.ID); err != nil {
+		return diag.Errorf("unable to create Transit gateway attachment (%s) for HVN (%s) and Transit gateway (%s): %v", tgwAtt.ID, tgwAtt.Hvn.ID, tgwAtt.ProviderData.AwsData.TgwID, err)
+	}
+
+	log.Printf("[INFO] Created Transit gateway attachment (%s) for HVN (%s) and Transit gateway (%s)", tgwAtt.ID, tgwAtt.Hvn.ID, tgwAtt.ProviderData.AwsData.TgwID)
+
+	// Set the globally unique id of this TGW attachment in the state now since
+	// it has been created, and from this point forward should be deletable
+	link := newLink(tgwAtt.Location, TgwAttachmentResourceType, tgwAtt.ID)
```

Comment:
We will want to move this block up above the `WaitForOperation` call, but holding off on changing this for now because https://github.com/hashicorp/terraform-provider-hcp/pull/59 is still in-progress.

Reply:
https://github.com/hashicorp/terraform-provider-hcp/pull/59 has been merged. +1 to moving the id setting above the operation polling

Reply:
Does that need to happen in any other resource or is this the last one?

Reply:
TGW should be the only resource right now that doesn't set the `id` before polling.

Reply:
Yep, it looked like Alex got all of the other cases of this in his PR. Will be updating this one to match